nx/source/services/set.c

```diff
@@ -720,11 +720,14 @@ Result setsysGetDeviceNickname(char* buffer) {
     return rc;
 }
 
-Result setsysSetDeviceNickname(const char* buffer) {
+Result setsysSetDeviceNickname(const char* nickname) {
+    char send_nickname[SET_MAX_NICKNAME_SIZE];
```

Comment:
Instead of doing an explicit memset, do `char send_nickname[SET_MAX_NICKNAME_SIZE] = {0};`